### PR TITLE
fix: cmp OOM

### PR DESCRIPTION
**Pull request by CraigMChen**
#### What this PR does / why we need it:

call cancel if steve server start failed

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sixther-dc @jferic 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | fix: call cancel if steve server start failed |
| 🇨🇳 中文    | 调用steve server的cancel函数以将其垃圾回收，修复cmp的OOM问题 |

### CODE REVIEW

File: modules/cmp/steve/aggregator.go [hunk 1]
```diff
@@ -208,10 +208,23 @@ func (a *Aggregator) Add(clusterInfo apistructs.ClusterInfo) {
 	a.servers.Store(clusterInfo.Name, g)
 	go func() {
 		logrus.Infof("starting steve server for cluster %s", clusterInfo.Name)
+		var err error
 		server, cancel, err := a.createSteve(clusterInfo)
+		defer func() {
+			if err != nil {
+				a.servers.Delete(clusterInfo.Name)
+				if cancel != nil {
```
Comment (jferic) [inline modules/cmp/steve/aggregator.go:216]: 先 cancel 再从 servers 里面 delete 是不是更好
delete 后应该就会触发gc了

**Review by sixther-dc — APPROVED** [review]
/approve

**Review by sixther-dc — COMMENTED** [review]
/approve

**Review by sixther-dc — APPROVED** [review]
/approve

**Pull request merged**